test/sharness/lib/test-generate-junit-html.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+
+dependencies=(
+  "url=https://sourceforge.net/projects/saxon/files/Saxon-HE/11/Java/SaxonHE11-4J.zip;md5=8a4783d307c32c898f8995b8f337fd6b"
+  "url=https://raw.githubusercontent.com/pl-strflt/ant/c781f7d79b92cc55530245d9554682a47f46851e/src/etc/junit-frames-saxon.xsl;md5=6eb013566903a91e4959413f6ff144d0"
+  "url=https://raw.githubusercontent.com/pl-strflt/ant/c781f7d79b92cc55530245d9554682a47f46851e/src/etc/junit-noframes-saxon.xsl;md5=8d54882d5f9d32a7743ec675cc2e30ac"
```

Comment:
We use https://saxon.sourceforge.net/#F11HE to turn JUnit XML output by sharness into HTMLs.

The XML-HTML transforms are defined in https://github.com/pl-strflt/ant/tree/feat/testcase-out/src/etc. We use a fork because the original didn't support `System.out` per testcase (among other things). I'll try to contribute `feat/testcase-out` back to https://github.com/apache/ant.

Reply:
nit: can we add `actions/cache` based on the hash of `test/sharness/lib/test-generate-junit-html.sh`, so download of these dependencies happens only once?

Reply:
Good idea! Done :)